packages/assets-controllers/src/TokenDetectionController.ts

```diff
@@ -701,26 +701,11 @@ export class TokenDetectionController extends StaticIntervalPollingController<To
 
     // Try detecting tokens via Account API first if conditions allow
     if (supportedNetworks && chainsToDetectUsingAccountAPI.length > 0) {
-      const apiResult = await this.#attemptAccountAPIDetection(
+      await this.#attemptAccountAPIDetection(
         chainsToDetectUsingAccountAPI,
         addressToDetect,
         supportedNetworks,
       );
-
-      // If API succeeds and no chains are left for RPC detection, we can return early
-      if (
-        apiResult?.result === 'success' &&
-        chainsToDetectUsingRpc.length === 0
-      ) {
-        return;
-      }
-
-      // If API fails or chainsToDetectUsingRpc still has items, add chains to RPC detection
-      this.#addChainsToRpcDetection(
```

Comment:
This seems like it was just adding all the account API chains, so it was doing an additional RPC based detection even though they just succeeded via account api.


If the account api call failed, we could try to fallback to RPC for all of the chains, although I don't think it's necessary so simplified.

Reply:
Let me know if either of these fixes makes sense to you @salimtb

Reply:
yes, i got the idea andthe fix makes sense to me. Just to confirm, given that the unit test is currently failing, with this fix, do we revert to the RPC method if the call to the account API fails ?

Reply:
I updated it to keep the RPC fallback if the account API request fails.